File: backend/SpringBoot2-Java/src/main/java/com/weweb/auth/model/UserContext.java
package com.weweb.auth.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserContext {

    private String username = "";
    private String[] roles = new String[0];

}

File: backend/Vertx3-Kotlin/src/main/kotlin/com/weweb/auth/service/UserContext.kt
package com.weweb.auth.service

/**
 * Context containing the profile of the caller of a protected end point
 *
 * @author Francesco Cina'
 */
class UserContext @JvmOverloads constructor(val username: String = "",
                                            val roles: Array<String> = arrayOf(),
                                            val isValid: Boolean = false)
